src/MRA.Jobs.Api/Application/Features/VacancyCategories/Queries/GetJobCategoriesQueryHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.EntityFrameworkCore;
+using MRA.Jobs.Application.Contracts.JobVacancies.Queries.GetJobCategories;
+using MRA.Jobs.Application.Contracts.JobVacancies.Responses;
+using MRA.Jobs.Application.Contracts.VacancyCategories.Responses;
+
+namespace MRA.Jobs.Application.Features.VacancyCategories.Queries;
+public class GetJobCategoriesQueryHandler : IRequestHandler<GetJobCategoriesQuery, List<JobCategoriesResponse>>
+{
+    private readonly IApplicationDbContext _context;
+    private readonly IMapper _mapper;
+
+    public GetJobCategoriesQueryHandler(IApplicationDbContext context, IMapper mapper)
+    {
+        _context = context;
+        _mapper = mapper;
+    }
+
+    public async Task<List<JobCategoriesResponse>> Handle(GetJobCategoriesQuery request, CancellationToken cancellationToken)
+    {
+        var jobs = (await _context.JobVacancies.ToListAsync()).AsEnumerable();
```

Comment:
You may remove ToListAsync.

var jobsQuery=_context.JobVacancies;

Reply:
I can’t delete, there is a condition, look at the code below, if there is an alternative solution, I’m waiting

Reply:
You may call tolist after building the query. This way you immediately get grouped and filtered data from the database

Reply:
how ?

Reply:
send me sample code please.